src/Octoshift/Services/GithubApi.cs

```diff
@@ -1089,24 +1090,102 @@ public virtual async Task<string> UploadArchiveToGithubStorage(string orgDatabas
+            return response;
         }
         else
         {
-            var url = $"https://uploads.github.com/organizations/{orgDatabaseId.EscapeDataString()}/gei/archive?name={archiveName.EscapeDataString()}";
+            var url = $"{_base_url}/{orgDatabaseId.EscapeDataString()}/gei/archive?name={archiveName.EscapeDataString()}";
 
             response = await _client.PostAsync(url, streamContent);
+            var data = JObject.Parse(response);
+            return (string)data["uri"];
         }
+    }
 
-        var data = JObject.Parse(response);
-        return (string)data["uri"];
+    private async Task<string> UploadMultipart(Stream archiveContent, string archiveName, string uploadUrl)
+    {
+        var buffer = new byte[_streamSizeLimit];
+
+        // 1. Start the upload
+        var (startResponse, startHeaders) = await StartUploadAsync(uploadUrl, archiveName, archiveContent.Length);
+
+        var nextUrl = GetNextUrl(startHeaders);
+        var guid = System.Web.HttpUtility.ParseQueryString(nextUrl.Query)["guid"];
+
+        // 2. Upload parts
+        int bytesRead;
+        while ((bytesRead = await archiveContent.ReadAsync(buffer, 0, buffer.Length)) > 0)
+        {
+            nextUrl = await UploadPartAsync(buffer, bytesRead, nextUrl.ToString());
+        }
+
+        // 3. Complete the upload
+        await CompleteUploadAsync(nextUrl.ToString());
+
+        return $"gei://archive/{guid}";
+    }
+
+    private async Task<(string Response, KeyValuePair<string, IEnumerable<string>>[] Headers)> StartUploadAsync(string uploadUrl, string archiveName, long contentSize)
+    {
+        // Create the body for the request
+        var body = new JObject
+        {
+            ["content_type"] = "application/octet-stream",
+            ["name"] = archiveName,
+            ["size"] = contentSize
+        };
+
+        // Post the request and get both response content and headers as a tuple
+        var (response, headers) = await _client.PostTupleAsync(uploadUrl, body);
+
+        // Return the tuple with response and headers
+        return (response, headers);
+    }
+
+
+    private async Task<Uri> UploadPartAsync(byte[] body, int bytesRead, string nextUrl)
+    {
+        var content = new ByteArrayContent(body, 0, bytesRead);
+        content.Headers.ContentType = new System.Net.Http.Headers.MediaTypeHeaderValue("application/octet-stream");
+        var (response, headers) = await _client.PatchTupleAsync(nextUrl, content);
+
+        return GetNextUrl(headers);
+    }
+
+    private async Task CompleteUploadAsync(string lastUrl)
+    {
+        // Create an empty HttpContent object
+        var content = new StringContent(string.Empty);
+
+        // Set the correct Content-Type header (adjust based on API requirements)
+        content.Headers.ContentType = new System.Net.Http.Headers.MediaTypeHeaderValue("application/octet-stream");
+
+        await _client.PutAsync(lastUrl, content);
```

Comment:
can we just do a simple put here:

```c#
await _client.PustAsync(lastUrl);
```

Reply:
Even in that case there is probably no need to create a string content because it's supposed to be handled by the client so you can probably just send an empty string to put.

Reply:
```
  public virtual async Task<string> PutAsync(string url, object body, Dictionary<string, string> customHeaders = null) =>
        (await SendAsync(HttpMethod.Put, url, body, customHeaders: customHeaders)).Content;
```

This is the put command, should I make a new one then?

Reply:
no you can pass `null` for `body`

Reply:
Ok so when I pass in `null` it will error out and not complete the upload. Adding this logic back in.